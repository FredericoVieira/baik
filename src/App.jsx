import React from 'react'

import Home from './screens/Home.screen'
import NavBar from './components/layout/NavBar.layout'

import AlertDialog from './components/dialogs/AlertDialog.component'
import Toast from './components/ui/Toast.component'

const App = () => (
  <>
    <NavBar />
    <Home />
    <AlertDialog />
    <Toast />
  </>
)

export default App
