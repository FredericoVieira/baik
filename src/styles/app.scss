@import url('https://fonts.googleapis.com/css2?family=Source+Sans+Pro:wght@300;400;700&display=swap');
@import url('https://fonts.googleapis.com/icon?family=Material+Icons');

*,
*::after,
*::before {
  margin: 0;
  padding: 0;
  box-sizing: inherit;
}

html {
  font-size: 62.5%;
}

body {
  box-sizing: border-box;
  overflow: auto !important;
  background-color: #020405;
  font-family: 'Source Sans Pro', sans-serif;
  font-weight: 300;
}

html,
body,
#app {
  height: 100%;
  display: flex;
  flex: 1;
  flex-direction: column;
  background: #020405;
}

section {
  padding: 100px;
}

.Toastify__toast {
  background: #353741;
  color: #fff;
  font-size: 1.5rem;
  padding: 15px;
  border-right: 5px solid #fff;
}

.Toastify__toast--success {
  border-color: #13c150;
}

.Toastify__toast--warning {
  border-color: #ff9300;
}

.Toastify__toast--error {
  border-color: #e10d05;
}

.mapboxgl-popup-close-button {
  padding: 15px;
  font-size: 20px;
  color: #fff;
}

.mapboxgl-popup-close-button:focus {
  outline: 0;
}

.mapboxgl-popup-anchor-top .mapboxgl-popup-tip {
  border-bottom-color: #353741 !important;
}

.mapboxgl-popup-content {
  background-color: #353741 !important;
  padding: 15px !important;
  min-width: 250px;
}
