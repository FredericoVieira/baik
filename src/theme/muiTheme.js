import { createMuiTheme } from '@material-ui/core/styles'

const theme = createMuiTheme({
  typography: {
    useNextVariants: true,
    fontSize: 18,
    fontFamily: ['Source Sans Pro']
  },
  palette: {
    primary: {
      main: '#F2AB1D'
    },
    secondary: {
      main: '#D64C1C'
    },
    info: {
      main: '#E42D95'
    },
    purple: {
      darkest: '#0B1113',
      darkness: '#131D20',
      dark: '#1F2E33',
      light: '#263A40'
    },
    gray: {
      darkness: '#353741',
      dark: '#747474',
      main: '#F2F2F2',
      light: '#E0E0E0'
    },
    auxiliary: {
      red: '#E10D05',
      green: '#13C150',
      white: '#FFF',
      black: '#000'
    },
    text: {
      primary: '#FFF',
      secondary: '#ABABAB',
      terciary: '#F2F2F2'
    }
  },
  overrides: {
    MuiButton: {
      contained: {
        '&Primary': {
          color: '#F2F2F2 !important'
        },
        '&Secondary': {
          color: '#F2F2F2 !important'
        },
        '&$disabled': {
          color: '#747474 !important',
          backgroundColor: '#ffffff1f'
        }
      },
      outlined: {
        color: '#F2F2F2',
        border: '1px solid rgba(255, 255, 255, 0.23)',
        '&:hover': {
          textDecoration: 'none',
          backgroundColor: 'inherit !important'
        },
        '&$disabled': {
          color: '#353741',
          border: '1px solid rgba(255, 255, 255, 0.23)'
        }
      }
    },
    MuiFormControlLabel: {
      label: {
        color: '#FFF'
      }
    },
    MuiFormLabel: {
      root: {
        '&$disabled': {
          color: '#747474'
        }
      }
    },
    MuiInput: {
      input: {
        '&:-webkit-autofill': {
          WebkitBoxShadow: '0 0 0 1000px #353741 inset',
          WebkitTextFillColor: '#FFF'
        }
      },
      underline: {
        '&::before': {
          borderBottom: '1px solid #F2F2F2'
        },
        '&$disabled': {
          color: '#747474'
        }
      }
    },
    MuiOutlinedInput: {
      notchedOutline: {
        borderColor: '#FFF'
      }
    },
    MuiPaper: {
      root: {
        backgroundColor: '#353741'
      }
    },
    MuiListItem: {
      root: {
        backgroundColor: '#353741',
        '&:hover': {
          backgroundColor: '#263A40 !important'
        },
        '& button ~ p': {
          marginRight: 12
        }
      }
    },
    MuiAppBar: {
      root: {
        boxShadow: 'none',
        backgroundColor: '#020405 !important'
      }
    },
    MuiTooltip: {
      tooltip: {
        fontSize: 12
      }
    },
    MuiTypography: {
      colorError: {
        color: '#E10D05'
      }
    }
  }
})

export default theme
