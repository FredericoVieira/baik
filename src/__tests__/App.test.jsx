import React from 'react'
import { render, screen } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core/styles'
import App from '../App'
import muiTheme from '../theme/muiTheme'

jest.mock('../utils/numbers.utils', () => ({
  randomIntFromInterval: jest.fn(() => 0)
}))

test('matches snapshot', () => {
  const { container } = render(
    <ThemeProvider theme={muiTheme}>
      <App />
    </ThemeProvider>
  )

  expect(container).toMatchSnapshot()
})

test('renders signin app screen', () => {
  render(
    <ThemeProvider theme={muiTheme}>
      <App />
    </ThemeProvider>
  )

  expect(screen.getByText('E-mail')).toBeInTheDocument()
  expect(screen.getByText('Senha')).toBeInTheDocument()
  expect(screen.getByText('Esqueceu sua senha?')).toBeInTheDocument()
  expect(screen.getByText('Sign In')).toBeInTheDocument()
})
