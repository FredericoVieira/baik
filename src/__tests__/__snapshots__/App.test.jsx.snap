// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular makeStyles-toolbar-1 MuiToolbar-gutters"
    >
      <div
        class="makeStyles-logoWrapper-2"
      >
        <img
          alt="baik-logo"
          class="makeStyles-logo-3"
          src="baik.png"
        />
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary"
        >
          Baik
        </h4>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6 MuiTypography-colorTextPrimary"
      >
        CONNECTING USERS TO BIKES
      </h6>
    </div>
  </header>
  <section>
    <div
      class="makeStyles-title-10"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary"
      >
        Bikes in all locations
      </h1>
      <h1
        class="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary"
      >
        for everyone
      </h1>
    </div>
    <div
      class="makeStyles-subtitle-11"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextSecondary"
      >
        Find your way to come and go wherever
      </h4>
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextSecondary"
      >
        and whenever you want
      </h4>
    </div>
    <div
      class="makeStyles-button-12"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Start biking for free
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </section>
  <section
    class="makeStyles-description-4"
  >
    <img
      alt="user-bikes"
      class="makeStyles-image-5"
      src="user-bikes.png"
    />
    <h2
      class="MuiTypography-root MuiTypography-h2 MuiTypography-colorTextPrimary"
    >
      We are a sustainable company that believes everyone can improve the planet. Go from Baik and collaborate with nature.
    </h2>
  </section>
  <section
    class="makeStyles-map-6"
  >
    <div
      style="position: relative; width: 100%; height: 70vh; cursor: grab; outline: none;"
      tabindex="0"
    >
      <div
        style="position: relative; width: 100%; height: 100%;"
      >
        <div
          class=""
          style="position: absolute; width: 100%; height: 100%; overflow: hidden; visibility: inherit;"
        />
      </div>
    </div>
    <div
      class="MuiPaper-root MuiAccordion-root makeStyles-controlPanel-14 MuiAccordion-rounded MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        aria-controls="panel1a-content"
        aria-disabled="false"
        aria-expanded="false"
        class="MuiButtonBase-root MuiAccordionSummary-root"
        id="panel1a-header"
        role="button"
        tabindex="0"
      >
        <div
          class="MuiAccordionSummary-content"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 MuiTypography-colorTextPrimary"
          >
            Networks by country
          </p>
        </div>
        <div
          aria-disabled="false"
          aria-hidden="true"
          class="MuiButtonBase-root MuiIconButton-root MuiAccordionSummary-expandIcon MuiIconButton-edgeEnd"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </div>
      </div>
      <div
        class="MuiCollapse-container MuiCollapse-hidden"
        style="min-height: 0px;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <div
              aria-labelledby="panel1a-header"
              id="panel1a-content"
              role="region"
            >
              <div
                class="MuiAccordionDetails-root makeStyles-networksByCountryWrapper-15"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="makeStyles-controlSelection-17"
    >
      <div
        class="makeStyles-pins-18"
      >
        <div
          class="makeStyles-pin-19"
        >
          <svg
            height="15"
            style="fill: #D64C1C; stroke: none;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.2,15.7L20.2,15.7c1.1-1.6,1.8-3.6,1.8-5.7c0-5.6-4.5-10-10-10S2,4.5,2,10c0,2,0.6,3.9,1.6,5.4c0,0.1,0.1,0.2,0.2,0.3
c0,0,0.1,0.1,0.1,0.2c0.2,0.3,0.4,0.6,0.7,0.9c2.6,3.1,7.4,7.6,7.4,7.6s4.8-4.5,7.4-7.5c0.2-0.3,0.5-0.6,0.7-0.9
C20.1,15.8,20.2,15.8,20.2,15.7z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Network
          </p>
        </div>
        <div
          class="makeStyles-pin-19"
        >
          <svg
            height="15"
            style="fill: #F2AB1D; stroke: none;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.2,15.7L20.2,15.7c1.1-1.6,1.8-3.6,1.8-5.7c0-5.6-4.5-10-10-10S2,4.5,2,10c0,2,0.6,3.9,1.6,5.4c0,0.1,0.1,0.2,0.2,0.3
c0,0,0.1,0.1,0.1,0.2c0.2,0.3,0.4,0.6,0.7,0.9c2.6,3.1,7.4,7.6,7.4,7.6s4.8-4.5,7.4-7.5c0.2-0.3,0.5-0.6,0.7-0.9
C20.1,15.8,20.2,15.8,20.2,15.7z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Selected network
          </p>
        </div>
        <div
          class="makeStyles-pin-19"
        >
          <svg
            height="15"
            style="fill: #88B73D; stroke: none;"
            viewBox="0 0 24 24"
          >
            <path
              d="M20.2,15.7L20.2,15.7c1.1-1.6,1.8-3.6,1.8-5.7c0-5.6-4.5-10-10-10S2,4.5,2,10c0,2,0.6,3.9,1.6,5.4c0,0.1,0.1,0.2,0.2,0.3
c0,0,0.1,0.1,0.1,0.2c0.2,0.3,0.4,0.6,0.7,0.9c2.6,3.1,7.4,7.6,7.4,7.6s4.8-4.5,7.4-7.5c0.2-0.3,0.5-0.6,0.7-0.9
C20.1,15.8,20.2,15.8,20.2,15.7z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextPrimary"
          >
            Station
          </p>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
        disabled=""
        tabindex="-1"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Clear selection
        </span>
      </button>
    </div>
  </section>
  <section
    class="makeStyles-repeat-7"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary"
    >
      Find your bike. Cycle to your destination. Give it back.
    </h4>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeLarge"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 7h10v3l4-4-4-4v3H5v6h2V7zm10 10H7v-3l-4 4 4 4v-3h12v-6h-2v4z"
      />
    </svg>
    <h4
      class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary"
    >
      Repeat.
    </h4>
  </section>
  <section>
    <div
      class="MuiPaper-root MuiCard-root makeStyles-card-21 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardContent-root makeStyles-cardContent-22"
      >
        <img
          alt="background"
          class="makeStyles-background-8"
          src="background.png"
        />
        <div
          class="makeStyles-anyDoubts-9"
        >
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-colorTextPrimary"
          >
            Any doubts?
          </h4>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
            tabindex="0"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              Contact us
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </section>
  <div
    class="Toastify"
  />
</div>
`;
